### PR TITLE
Use declaring class as qualifier

### PR DESCRIPTION
**Pull request by basil**
Applied an automated Eclipse refactoring to change indirect accesses (through subtypes) to static members to direct accesses. I find the code easier to read when there is a consistent way of referring to a given member.

### CODE REVIEW

File: core/src/main/java/hudson/model/FingerprintCleanupThread.java [hunk 1]
```diff
@@ -79,7 +79,7 @@ public void execute(TaskListener listener) {
 
         if (!(FingerprintStorage.get() instanceof FileFingerprintStorage) &&
                 FingerprintStorage.getFileFingerprintStorage().isReady()) {
-            FileFingerprintStorage.getFileFingerprintStorage().iterateAndCleanupFingerprints(listener);
+            FingerprintStorage.getFileFingerprintStorage().iterateAndCleanupFingerprints(listener);
```
Comment (daniel-beck) [inline core/src/main/java/hudson/model/FingerprintCleanupThread.java:82]: This PR already provides benefits: Revealing an obvious bad name in the API.

https://github.com/jenkinsci/jenkins/blob/6f2e7b0085e39d02a92312a57a5cf7f0b84dfc05/core/src/main/java/jenkins/fingerprints/FingerprintStorage.java#L61-L63

File: core/src/main/java/hudson/model/Slave.java [hunk 2]
```diff
@@ -308,7 +308,7 @@ public void setNodeProperties(List<? extends NodeProperty<?>> properties) throws
     }
 
     public RetentionStrategy getRetentionStrategy() {
-        return retentionStrategy == null ? RetentionStrategy.Always.INSTANCE : retentionStrategy;
+        return retentionStrategy == null ? RetentionStrategy.INSTANCE : retentionStrategy;
```
Comment (daniel-beck) [inline core/src/main/java/hudson/model/Slave.java:311]: Due to the bad name of the constant, this is more confusing now.

File: test/src/test/java/hudson/model/ProjectTest.java [hunk 3]
```diff
@@ -608,9 +608,9 @@ public void testDoDoWipeOutWorkspace() throws Exception{
                fail("AccessDeniedException should be thrown.");
             }
         } 
-        auth.add(Job.READ, user.getId());
-        auth.add(Job.BUILD, user.getId());
-        auth.add(Job.WIPEOUT, user.getId());
+        auth.add(Item.READ, user.getId());
+        auth.add(Item.BUILD, user.getId());
```
Comment (daniel-beck) [inline test/src/test/java/hudson/model/ProjectTest.java:612]: 😭

File: core/src/main/java/jenkins/model/GlobalProjectNamingStrategyConfiguration.java [hunk 4]
```diff
@@ -58,7 +57,7 @@ public boolean configure(StaplerRequest req, JSONObject json) throws hudson.mode
             }
         }
         if (j.getProjectNamingStrategy() == null) {
-            j.setProjectNamingStrategy(DefaultProjectNamingStrategy.DEFAULT_NAMING_STRATEGY);
+            j.setProjectNamingStrategy(ProjectNamingStrategy.DEFAULT_NAMING_STRATEGY);
```
Comment (daniel-beck) [inline core/src/main/java/jenkins/model/GlobalProjectNamingStrategyConfiguration.java:60]: Also weird and should be cleaned up 😭

**Review by daniel-beck — APPROVED** [review]
Overall seems fine, although I would argue the change to `Slave.java` makes it less readable.

File: core/src/main/java/hudson/model/Slave.java (hunk 2 above)
Comment (basil) [reply core/src/main/java/hudson/model/Slave.java:311]: Good point. I'll revert this change.

Comment (daniel-beck) [reply core/src/main/java/hudson/model/Slave.java]: Another option could be to change `RetentionStategy`, deprecating `INSTANCE` and either replacing all its callers with `new RetentionStategy.Always()`, or a new `ALWAYS` constant whose name is not ambiguous.

**Review by daniel-beck — APPROVED** [review]

**Review by timja — APPROVED** [review]

**Comment by timja** [comment]
This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

**Review by oleg-nenashev — APPROVED** [review]
LGTM

**Comment by oleg-nenashev** [comment]
sorry for not squash-merging ....

**Pull request merged**